pkg/k8s/utils/factory.go

```diff
@@ -45,8 +46,35 @@ var (
 	casts = map[reflect.Type]castToDeepCopy{}
 	// resourcers maps an interface to a resource name (string).
 	resourcers = map[reflect.Type]string{}
+
+	// lastEventReceived is the time at which the last k8s event was received
+	lastEventReceived eventTimestamper
```

Comment:
Consider removing this change and adding this in the metric callback of k8s_watcher.go. My PR #7301 will remove this file altogether.

Reply:
I've pushed the code into `pkg/k8s/metrics` and trigger the reset from all apiserver calls now so we can just remove the event handler when that code goes away.